src/App.tsx

```diff
@@ -15,48 +16,128 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort((g1, g2) => g1.localeCompare(g2));
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((g1, g2) => g1.length - g2.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReverced] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const myPrepareProducts = getReorderedGoods(
+    goodsFromServer,
+    { sortType, isReversed },
+  );
+  const visibleReset = isReversed === true || (sortType !== SortType.NONE);
```

Comment:
```suggestion
  const isResetVisiblet = isReversed === true || (sortType !== SortType.NONE);
```
isResetVisible or shouldResetButtonRender
the condition should start from` is` or `should`

Reply:
done